src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -590,7 +673,7 @@ typename MatType::elem_type RNN<
       // TODO: note that we could avoid the copy of currentGradient by having
       // each layer *add* its gradient to `gradient`.  However that would
       // require some amount of refactoring.
-      MatType networkDelta;
+      MatType networkDelta(predictors.n_rows, batchSize);
```

Comment:
Nice catch---I think not every layer assumes that `networkDelta` needs to be resized.  You could probably add `arma::fill::none` here if we can assume that every layer is going to overwrite `networkDelta` (I think that is a safe assumption).